models/repo.go

```diff
@@ -811,11 +811,15 @@ func MigrateRepository(u *User, opts MigrateRepoOptions) (*Repository, error) {
 	}
 
 	if opts.IsMirror {
+		duration, err := time.ParseDuration(setting.Mirror.DefaultInterval)
```

Comment:
Also, just a note, time.ParseDuration accepts negative durations. This should be checked for.